### ISSUE TITLE
[JENKINS-11256] Maven3 parallel build fails with java.util.ConcurrentModificationException in Jenkins

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-23#4815]
<p>linux, jenkins 1.433, maven 3.0.3, maven project with maven 3 and -T 8 running on a remote ssh slave<br/>
build fails with</p>

<p>&#8212; maven-install-plugin:2.3.1:install (default-install) @ xxxxxxxx &#8212;<br/>
Waiting for Jenkins to finish collecting data<br/>
mavenExecutionResult exceptions not empty<br/>
message : Internal error: java.util.ConcurrentModificationException<br/>
cause : null<br/>
Stack trace : <br/>
org.apache.maven.InternalErrorException: Internal error: java.util.ConcurrentModificationException<br/>
	at org.apache.maven.lifecycle.internal.BuilderCommon.handleBuildError(BuilderCommon.java:128)<span class="error">&#91;INFO&#93;</span> Installing xxxxxxxx.jar to xxxxxxxx</p>

<p>	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:95)<br/>
	at org.apache.maven.lifecycle.internal.LifecycleThreadedBuilder$1.call(LifecycleThreadedBuilder.java:167)<br/>
	at org.apache.maven.lifecycle.internal.LifecycleThreadedBuilder$1.call(LifecycleThreadedBuilder.java:164)<br/>
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:269)<br/>
	at java.util.concurrent.FutureTask.run(FutureTask.java:123)<span class="error">&#91;INFO&#93;</span> Compiling 1 source file to xxxxxxxx</p>

<p>	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:651)<br/>
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:676)<br/>
	at java.lang.Thread.run(Thread.java:595)<br/>
Caused by: java.util.ConcurrentModificationException<br/>
	at java.util.ArrayList.writeObject(ArrayList.java:572)<br/>
	at sun.reflect.GeneratedMethodAccessor15.invoke(Unknown Source)<br/>
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)<br/>
	at java.lang.reflect.Method.invoke(Method.java:592)<br/>
	at java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:917)<br/>
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1339)<br/>
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1290)<br/>
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1079)<br/>
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1375)<br/>
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1347)<br/>
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1290)<br/>
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1079)<br/>
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1375)<br/>
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1347)<br/>
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1290)<br/>
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1079)<br/>
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:302)<br/>
	at java.util.ArrayList.writeObject(ArrayList.java:569)<br/>
	at sun.reflect.GeneratedMethodAccessor15.invoke(Unknown Source)<br/>
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)<br/>
	at java.lang.reflect.Method.invoke(Method.java:592)<br/>
	at java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:917)<br/>
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1339)<br/>
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1290)<br/>
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1079)<br/>
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:302)<br/>
	at java.util.HashMap.writeObject(HashMap.java:1039)<br/>
	at sun.reflect.GeneratedMethodAccessor36.invoke(Unknown Source)<br/>
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)<br/>
	at java.lang.reflect.Method.invoke(Method.java:592)<br/>
	at java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:917)<br/>
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1339)<br/>
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1290)<br/>
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1079)<br/>
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1375)<br/>
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1347)<br/>
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1290)<br/>
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1079)<br/>
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1375)<br/>
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1347)<br/>
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1290)<br/>
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1079)<br/>
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1375)<br/>
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1347)<br/>
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1290)<br/>
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1079)<br/>
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:302)<br/>
	at hudson.remoting.UserRequest._serialize(UserRequest.java:155)<br/>
	at hudson.remoting.UserRequest.serialize(UserRequest.java:164)<br/>
	at hudson.remoting.UserRequest.<init>(UserRequest.java:62)<br/>
	at hudson.remoting.Channel.callAsync(Channel.java:708)<br/>
	at hudson.maven.Maven3Builder$FilterImpl.executeAsync(Maven3Builder.java:229)<br/>
	at hudson.maven.reporters.MavenFingerprinter.postBuild(MavenFingerprinter.java:96)<br/>
	at hudson.maven.Maven3Builder$MavenExecutionListener.recordProjectSucceeded(Maven3Builder.java:431)<br/>
	at hudson.maven.Maven3Builder$MavenExecutionListener.projectSucceeded(Maven3Builder.java:405)<br/>
	at org.apache.maven.lifecycle.internal.DefaultExecutionEventCatapult.fire(DefaultExecutionEventCatapult.java:74)<br/>
	at org.apache.maven.lifecycle.internal.DefaultExecutionEventCatapult.fire(DefaultExecutionEventCatapult.java:42)<br/>
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:91)<br/>
	... 7 more<br/>
<span class="error">&#91;INFO&#93;</span> <br/>
<span class="error">&#91;INFO&#93;</span> &#8212; maven-compiler-plugin:2.3.2:testCompile (default-testCompile) @ xxxxxxxx &#8212;<br/>
channel stopped<br/>
Archiviere Artefakte<br/>
Finished: FAILURE</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kisa">kisa</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-11256" target="_blank">Maven3 parallel build fails with java.util.ConcurrentModificationException in Jenkins</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kutzi.jpg" title="kutzi's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kutzi">kutzi</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>component(s)</b>: core
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2011-11-18T14:20:29+00:00
<li><b>votes</b>: 2
<li><b>watchers</b>: 8
<li><b>imported</b>: 2025-11-23
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>linux, jenkins 1.433, maven 3.0.3, maven project with maven 3 and -T 8 running on a remote ssh slave
build fails with</p>

<p>&#8212; maven-install-plugin:2.3.1:install (default-install) @ xxxxxxxx &#8212;
Waiting for Jenkins to finish collecting data
mavenExecutionResult exceptions not empty
message : Internal error: java.util.ConcurrentModificationException
cause : null
Stack trace : 
org.apache.maven.InternalErrorException: Internal error: java.util.ConcurrentModificationException
	at org.apache.maven.lifecycle.internal.BuilderCommon.handleBuildError(BuilderCommon.java:128)<span class="error">&#91;INFO&#93;</span> Installing xxxxxxxx.jar to xxxxxxxx</p>

<p>	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:95)
	at org.apache.maven.lifecycle.internal.LifecycleThreadedBuilder$1.call(LifecycleThreadedBuilder.java:167)
	at org.apache.maven.lifecycle.internal.LifecycleThreadedBuilder$1.call(LifecycleThreadedBuilder.java:164)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:269)
	at java.util.concurrent.FutureTask.run(FutureTask.java:123)<span class="error">&#91;INFO&#93;</span> Compiling 1 source file to xxxxxxxx</p>

<p>	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:651)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:676)
	at java.lang.Thread.run(Thread.java:595)
Caused by: java.util.ConcurrentModificationException
	at java.util.ArrayList.writeObject(ArrayList.java:572)
	at sun.reflect.GeneratedMethodAccessor15.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:592)
	at java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:917)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1339)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1290)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1079)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1375)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1347)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1290)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1079)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1375)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1347)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1290)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1079)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:302)
	at java.util.ArrayList.writeObject(ArrayList.java:569)
	at sun.reflect.GeneratedMethodAccessor15.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:592)
	at java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:917)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1339)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1290)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1079)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:302)
	at java.util.HashMap.writeObject(HashMap.java:1039)
	at sun.reflect.GeneratedMethodAccessor36.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:592)
	at java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:917)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1339)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1290)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1079)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1375)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1347)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1290)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1079)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1375)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1347)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1290)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1079)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1375)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1347)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1290)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1079)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:302)
	at hudson.remoting.UserRequest._serialize(UserRequest.java:155)
	at hudson.remoting.UserRequest.serialize(UserRequest.java:164)
	at hudson.remoting.UserRequest.&lt;init&gt;(UserRequest.java:62)
	at hudson.remoting.Channel.callAsync(Channel.java:708)
	at hudson.maven.Maven3Builder$FilterImpl.executeAsync(Maven3Builder.java:229)
	at hudson.maven.reporters.MavenFingerprinter.postBuild(MavenFingerprinter.java:96)
	at hudson.maven.Maven3Builder$MavenExecutionListener.recordProjectSucceeded(Maven3Builder.java:431)
	at hudson.maven.Maven3Builder$MavenExecutionListener.projectSucceeded(Maven3Builder.java:405)
	at org.apache.maven.lifecycle.internal.DefaultExecutionEventCatapult.fire(DefaultExecutionEventCatapult.java:74)
	at org.apache.maven.lifecycle.internal.DefaultExecutionEventCatapult.fire(DefaultExecutionEventCatapult.java:42)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:91)
	... 7 more
<span class="error">&#91;INFO&#93;</span> 
<span class="error">&#91;INFO&#93;</span> &#8212; maven-compiler-plugin:2.3.2:testCompile (default-testCompile) @ xxxxxxxx &#8212;
channel stopped
Archiviere Artefakte
Finished: FAILURE</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>linux, jenkins 1.433, maven 3.0.3, maven project with maven 3 and -T 8 running on a remote ssh slave</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kutzi.jpg" title="kutzi's avatar" /> kutzi:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-11256?focusedId=153723&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-153723">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Does it happen frequently or only once?</p>
</pre>
</details>

</ul>
</details>

<p>Does it happen frequently or only once?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/paux.jpg" title="paux's avatar" /> paux:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-11256?focusedId=153776&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-153776">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>All three collections in Maven Fingerprinter need to be synchronized (Collections.synchronized...()) schould be sufficient. This problem could occur at several places.</p>

<p>Also, check wether a given file has alread been digested should be converted into a onestep checkAndSet operation.</p>

<p>I will create a pull request for this.</p>
</pre>
</details>

</ul>
</details>

<p>All three collections in Maven Fingerprinter need to be synchronized (Collections.synchronized...()) schould be sufficient. This problem could occur at several places.</p>

<p>Also, check wether a given file has alread been digested should be converted into a onestep checkAndSet operation.</p>

<p>I will create a pull request for this.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kutzi.jpg" title="kutzi's avatar" /> kutzi:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-11256?focusedId=153781&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-153781">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks for the pull request.
Yes, please create a separate pull request for each issue.</p>

<p>Also, I wonder if the 3 maps/sets are really the problem, as they're (as far as I understand it, are per Maven module and therefore only written in a single thread). My suspicion was much more regarding the FingerprintAction which is added and modified in a non-threadsafe fashion. Have you tried if your change fixes the problem?</p>

<p>I think I'd rather put a synchronized block inside the BuildCallable in postBuild()</p>
</pre>
</details>

</ul>
</details>

<p>Thanks for the pull request.<br/>
Yes, please create a separate pull request for each issue.</p>

<p>Also, I wonder if the 3 maps/sets are really the problem, as they're (as far as I understand it, are per Maven module and therefore only written in a single thread). My suspicion was much more regarding the FingerprintAction which is added and modified in a non-threadsafe fashion. Have you tried if your change fixes the problem?</p>

<p>I think I'd rather put a synchronized block inside the BuildCallable in postBuild()</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/paux.jpg" title="paux's avatar" /> paux:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-11256?focusedId=155553&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-155553">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>@kutzi: My development environment seems broken. Could you do that change?</p>

<p>Thx</p>
</pre>
</details>

</ul>
</details>

<p>@​kutzi: My development environment seems broken. Could you do that change?</p>

<p>Thx</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kutzi.jpg" title="kutzi's avatar" /> kutzi:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-11256?focusedId=155574&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-155574">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>While it might make sense to make TestResult thread-safe, I'd also ask if it's actually needed that SurefireArchiver#result is serialized.
Maybe that could be changed, too?</p>
</pre>
</details>

</ul>
</details>

<p>While it might make sense to make TestResult thread-safe, I'd also ask if it's actually needed that SurefireArchiver#result is serialized.<br/>
Maybe that could be changed, too?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-11256?focusedId=155602&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-155602">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Christoph Kutzinski
Path:
 maven-plugin/src/main/java/hudson/maven/reporters/SurefireArchiver.java
<a href="http://jenkins-ci.org/commit/jenkins/1476b2ee446f27b0c4ce85c16c5f984ef84cc8a4" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/1476b2ee446f27b0c4ce85c16c5f984ef84cc8a4</a>
Log:
  <span class="error">&#91;FIXED JENKINS-11256&#93;</span> SurefireArchiver#result doesn't need to be serialized on each remote call. Fixes ConcurrentModificationExceptions in MavenFingerprinter</p>


<p>Compare: <a href="https://github.com/jenkinsci/jenkins/compare/796918d...1476b2e" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/796918d...1476b2e</a></p>

</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Christoph Kutzinski<br/>
Path:<br/>
 maven-plugin/src/main/java/hudson/maven/reporters/SurefireArchiver.java<br/>
<a href="http://jenkins-ci.org/commit/jenkins/1476b2ee446f27b0c4ce85c16c5f984ef84cc8a4" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/1476b2ee446f27b0c4ce85c16c5f984ef84cc8a4</a><br/>
Log:<br/>
  <span class="error">&#91;FIXED JENKINS-11256&#93;</span> SurefireArchiver#result doesn't need to be serialized on each remote call. Fixes ConcurrentModificationExceptions in MavenFingerprinter</p>


<p>Compare: <a href="https://github.com/jenkinsci/jenkins/compare/796918d...1476b2e" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/compare/796918d...1476b2e</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-11256?focusedId=155604&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-155604">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Christoph Kutzinski
Path:
 changelog.html
<a href="http://jenkins-ci.org/commit/jenkins/b835fb15497cf31a5069804a9920a747053be3c4" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/b835fb15497cf31a5069804a9920a747053be3c4</a>
Log:
  Changelog entry for <a href="https://issues.jenkins.io/browse/JENKINS-11256" title="Maven3 parallel build fails with java.util.ConcurrentModificationException in Jenkins" class="issue-link" data-issue-key="JENKINS-11256"><del>JENKINS-11256</del></a></p>



</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Christoph Kutzinski<br/>
Path:<br/>
 changelog.html<br/>
<a href="http://jenkins-ci.org/commit/jenkins/b835fb15497cf31a5069804a9920a747053be3c4" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/b835fb15497cf31a5069804a9920a747053be3c4</a><br/>
Log:<br/>
  Changelog entry for <a href="https://issues.jenkins.io/browse/JENKINS-11256" title="Maven3 parallel build fails with java.util.ConcurrentModificationException in Jenkins" class="issue-link" data-issue-key="JENKINS-11256"><del>JENKINS-11256</del></a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/dogfood.png" title="dogfood's avatar" /> dogfood:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-11256?focusedId=155614&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-155614">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/yellow.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/1303/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #1303</a>
     <span class="error">&#91;FIXED JENKINS-11256&#93;</span> SurefireArchiver#result doesn't need to be serialized on each remote call. Fixes ConcurrentModificationExceptions in MavenFingerprinter
Changelog entry for <a href="https://issues.jenkins.io/browse/JENKINS-11256" title="Maven3 parallel build fails with java.util.ConcurrentModificationException in Jenkins" class="issue-link" data-issue-key="JENKINS-11256"><del>JENKINS-11256</del></a></p>

<p>Christoph Kutzinski : <a href="https://github.com/jenkinsci/jenkins/commit/1476b2ee446f27b0c4ce85c16c5f984ef84cc8a4" class="external-link" target="_blank" rel="nofollow noopener">1476b2ee446f27b0c4ce85c16c5f984ef84cc8a4</a>
Files : </p>
<ul>
	<li>maven-plugin/src/main/java/hudson/maven/reporters/SurefireArchiver.java</li>
</ul>


<p>Christoph Kutzinski : <a href="https://github.com/jenkinsci/jenkins/commit/b835fb15497cf31a5069804a9920a747053be3c4" class="external-link" target="_blank" rel="nofollow noopener">b835fb15497cf31a5069804a9920a747053be3c4</a>
Files : </p>
<ul>
	<li>changelog.html</li>
</ul>

</pre>
</details>

</ul>
</details>

<p>Integrated in <span class="image-wrap" style=""><img src="http://ci.jenkins-ci.org/images/16x16/yellow.png" style="border: 0px solid black" /></span> <a href="http://ci.jenkins-ci.org/job/jenkins_main_trunk/1303/" class="external-link" target="_blank" rel="nofollow noopener">jenkins_main_trunk #1303</a><br/>
     <span class="error">&#91;FIXED JENKINS-11256&#93;</span> SurefireArchiver#result doesn't need to be serialized on each remote call. Fixes ConcurrentModificationExceptions in MavenFingerprinter<br/>
Changelog entry for <a href="https://issues.jenkins.io/browse/JENKINS-11256" title="Maven3 parallel build fails with java.util.ConcurrentModificationException in Jenkins" class="issue-link" data-issue-key="JENKINS-11256"><del>JENKINS-11256</del></a></p>

<p>Christoph Kutzinski : <a href="https://github.com/jenkinsci/jenkins/commit/1476b2ee446f27b0c4ce85c16c5f984ef84cc8a4" class="external-link" target="_blank" rel="nofollow noopener">1476b2ee446f27b0c4ce85c16c5f984ef84cc8a4</a><br/>
Files : </p>
<ul>
	<li>maven-plugin/src/main/java/hudson/maven/reporters/SurefireArchiver.java</li>
</ul>


<p>Christoph Kutzinski : <a href="https://github.com/jenkinsci/jenkins/commit/b835fb15497cf31a5069804a9920a747053be3c4" class="external-link" target="_blank" rel="nofollow noopener">b835fb15497cf31a5069804a9920a747053be3c4</a><br/>
Files : </p>
<ul>
	<li>changelog.html</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>swolk:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-11256?focusedId=156791&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-156791">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>When will this make it into an LTS build? I really need this fix. Thanks</p>
</pre>
</details>

</ul>
</details>

<p>When will this make it into an LTS build? I really need this fix. Thanks</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-11256?focusedId=156850&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-156850">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Christoph Kutzinski
Path:
 maven-plugin/src/main/java/hudson/maven/reporters/SurefireArchiver.java
<a href="http://jenkins-ci.org/commit/jenkins/46629db8d493f67c5b8ce1c14465b3c6a117e847" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/46629db8d493f67c5b8ce1c14465b3c6a117e847</a>
Log:
  <span class="error">&#91;FIXED JENKINS-11256&#93;</span> SurefireArchiver#result doesn't need to be serialized on each remote call. Fixes ConcurrentModificationExceptions in MavenFingerprinter(cherry picked from commit 1476b2ee446f27b0c4ce85c16c5f984ef84cc8a4)</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Christoph Kutzinski<br/>
Path:<br/>
 maven-plugin/src/main/java/hudson/maven/reporters/SurefireArchiver.java<br/>
<a href="http://jenkins-ci.org/commit/jenkins/46629db8d493f67c5b8ce1c14465b3c6a117e847" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/46629db8d493f67c5b8ce1c14465b3c6a117e847</a><br/>
Log:<br/>
  <span class="error">&#91;FIXED JENKINS-11256&#93;</span> SurefireArchiver#result doesn't need to be serialized on each remote call. Fixes ConcurrentModificationExceptions in MavenFingerprinter(cherry picked from commit 1476b2ee446f27b0c4ce85c16c5f984ef84cc8a4)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-11256?focusedId=156851&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-156851">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Christoph Kutzinski
Path:
 changelog.html
<a href="http://jenkins-ci.org/commit/jenkins/b39a310c5165234dab6d7acd3ed82bd399d255eb" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/b39a310c5165234dab6d7acd3ed82bd399d255eb</a>
Log:
  Changelog entry for <a href="https://issues.jenkins.io/browse/JENKINS-11256" title="Maven3 parallel build fails with java.util.ConcurrentModificationException in Jenkins" class="issue-link" data-issue-key="JENKINS-11256"><del>JENKINS-11256</del></a>(cherry picked from commit b835fb15497cf31a5069804a9920a747053be3c4)</p>

<p>Conflicts:</p>

<p>	changelog.html</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Christoph Kutzinski<br/>
Path:<br/>
 changelog.html<br/>
<a href="http://jenkins-ci.org/commit/jenkins/b39a310c5165234dab6d7acd3ed82bd399d255eb" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/jenkins/b39a310c5165234dab6d7acd3ed82bd399d255eb</a><br/>
Log:<br/>
  Changelog entry for <a href="https://issues.jenkins.io/browse/JENKINS-11256" title="Maven3 parallel build fails with java.util.ConcurrentModificationException in Jenkins" class="issue-link" data-issue-key="JENKINS-11256"><del>JENKINS-11256</del></a>(cherry picked from commit b835fb15497cf31a5069804a9920a747053be3c4)</p>

<p>Conflicts:</p>

<p>	changelog.html</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-11256?focusedId=186653&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-186653">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Christoph Kutzinski
Path:
 src/main/java/hudson/maven/reporters/SurefireArchiver.java
<a href="http://jenkins-ci.org/commit/maven-plugin/00f1af0b4e71f8c691e36a1bb741eeff7bd512ee" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/maven-plugin/00f1af0b4e71f8c691e36a1bb741eeff7bd512ee</a>
Log:
  <span class="error">&#91;FIXED JENKINS-11256&#93;</span> SurefireArchiver#result doesn't need to be serialized on each remote call. Fixes ConcurrentModificationExceptions in MavenFingerprinter
Originally-Committed-As: 1476b2ee446f27b0c4ce85c16c5f984ef84cc8a4</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Christoph Kutzinski<br/>
Path:<br/>
 src/main/java/hudson/maven/reporters/SurefireArchiver.java<br/>
<a href="http://jenkins-ci.org/commit/maven-plugin/00f1af0b4e71f8c691e36a1bb741eeff7bd512ee" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/maven-plugin/00f1af0b4e71f8c691e36a1bb741eeff7bd512ee</a><br/>
Log:<br/>
  <span class="error">&#91;FIXED JENKINS-11256&#93;</span> SurefireArchiver#result doesn't need to be serialized on each remote call. Fixes ConcurrentModificationExceptions in MavenFingerprinter<br/>
Originally-Committed-As: 1476b2ee446f27b0c4ce85c16c5f984ef84cc8a4</p>
